pandasai/llm/google_vertexai.py

```diff
@@ -127,7 +127,8 @@ def _generate_text(self, prompt: str) -> str:
                 max_output_tokens=self.max_output_tokens,
             )
         else:
-            raise UnsupportedModelError("Unsupported model")
+            raise UnsupportedModelError(self.model)
```

Comment:
@arian81 also please consider using the exactly name of a class instead of it's representation, make it to be something like this:
```python
raise UnsupportedModelError(self.model.__name__)
```
Just because we've got already the same like [here](https://github.com/gventuri/pandas-ai/blob/720811f9ebd5632a20aeff01ea7b73f125ba3a48/pandasai/prompts/file_based_prompt.py#L34).

Of course, this isn't important at all, it's on your choice, both of the ways look good as for me.

Reply:
I see that they used the other notation elsewhere but in the same logic code they're using self.model in the if statements. To me using other notation only for the exception looks inconsistent. Maybe there's a reasoning for using that notation in the other place, @gventuri can you elaborate on this ? 
```py
if self.model in self._supported_code_models:
            code_generation = CodeGenerationModel.from_pretrained(self.model)

            completion = code_generation.predict(
                prefix=prompt,
                temperature=self.temperature,
                max_output_tokens=self.max_output_tokens,
            )
        elif self.model in self._supported_text_models:
            text_generation = TextGenerationModel.from_pretrained(self.model)

            completion = text_generation.predict(
                prompt=prompt,
                temperature=self.temperature,
                top_p=self.top_p,
                top_k=self.top_k,
                max_output_tokens=self.max_output_tokens,
            )
        else:
            raise UnsupportedModelError(self.model)```

Reply:
@nautics889 thanks a lot for reporting, it's actually a very good point. I think in the other implementation, we were forced, since the template itself doesn't have a name, while the model does.
In this case, we want the specific model (i.e. code-bison) to be provided, as opposed to the LLM class.